lib/pbio/sys/program_load.c

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022 The Pybricks Authors
+
+#include <pbsys/config.h>
+
+#if PBSYS_CONFIG_PROGRAM_LOAD
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+
+#include <contiki.h>
+
+#include <pbio/button.h>
+#include <pbio/main.h>
+#include <pbio/math.h>
+#include <pbio/protocol.h>
+
+#include <pbsys/bluetooth.h>
+#include <pbsys/main.h>
+#include <pbsys/status.h>
+
+/**
+ * Map of stored data.
+ */
+typedef struct {
+    /**
+     * Size of the application program.
+     */
+    uint32_t program_size;
+    /**
+     * Data of the application program.
+     */
+    uint8_t program_data[];
+} __attribute__((scalar_storage_order("little-endian"))) data_map_t;
```

Comment:
But that's kind of what it is for, right? It doesn't hurt now and would ensure correctness if it was ever used on big endian.

But it was perhaps more so to make the code self explanatory, since this stuff is received over the air. This way, someone writing an editor or tool would know what to send.

IIRC pybricks code has a comment that says this is big endian even though it isn't. 

I don't mind removing it though :)

Reply:
https://github.com/pybricks/pybricks-code/blob/master/src/hub/sagas.ts#L84

Reply:
Removed.

Reply:
I've never actually seen it used before. Reading the GCC docs, it seems like it comes with quite a bit of restrictions.